r/tests/testthat/test-dplyr-query.R

```diff
@@ -119,7 +119,7 @@ test_that("collect(as_data_frame=FALSE)", {
     filter(int > 5) %>%
     group_by(int) %>%
     collect(as_data_frame = FALSE)
-  expect_s3_class(b4, "arrow_dplyr_query")
```

Comment:
I think we need a test (and matching code change) that a Table with groups in the `$metadata` prints that it is grouped. Currently, the print method does not reveal that you have groupings, even though they get restored when you `as.data.frame()`:

```
> tab <- arrow_table(mtcars)
> tab$metadata$r$attributes$.group_vars <- "cyl"
> tab
Table
32 rows x 11 columns
$mpg <double>
$cyl <double>
$disp <double>
$hp <double>
$drat <double>
$wt <double>
$qsec <double>
$vs <double>
$am <double>
$gear <double>
$carb <double>

See $metadata for additional Schema metadata
```

I wonder what else is lost or concealed by this change. This feature of reading groups from `$metadata$r$attributes$.group_vars` was a bit of a hack and not something we're generally relying on, so I don't think it's a simple change. Or at least, it's not something we can just rely that the test suite already covers everything we care about. (Search for ".group_vars" in the code, it's not touched in many places, and only in one test.)

Not a reason not to do it, just need to be thorough.

Reply:
Thanks for your review.

> I think we need a test (and matching code change) that a Table with groups in the `$metadata` prints that it is grouped. Currently, the print method does not reveal that you have groupings, even though they get restored when you `as.data.frame()`:

I agree that the print results need improvement.
But would it be better to create a separate ticket and do it there?
This occurs even in arrow 9.0.0, regardless of the behavior with `compute`.

``` r
mtcars |> dplyr::group_by(cyl) |> arrow::arrow_table()
#> Table
#> 32 rows x 11 columns
#> $mpg <double>
#> $cyl <double>
#> $disp <double>
#> $hp <double>
#> $drat <double>
#> $wt <double>
#> $qsec <double>
#> $vs <double>
#> $am <double>
#> $gear <double>
#> $carb <double>
#>
#> See $metadata for additional Schema metadata
```

<sup>Created on 2022-09-20 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

> This feature of reading groups from `$metadata$r$attributes$.group_vars` was a bit of a hack and not something we're generally relying on, so I don't think it's a simple change. Or at least, it's not something we can just rely that the test suite already covers everything we care about.

Since `dplyr::group_vars` reads this field, isn't this function generally used?